bga_database/base_models.py

```diff
@@ -1,8 +1,36 @@
 from django.db import models
+from django.db.models import Q
 
 
 class SluggedModel(models.Model):
     slug = models.SlugField(max_length=255, unique=True, null=True)
 
     class Meta:
         abstract = True
+
+
+class AliasModel(models.Model):
+    name = models.CharField(max_length=255)
+    preferred = models.BooleanField(default=False)
+
+    class Meta:
+        abstract = True
+
+    def __str__(self):
+        return self.name
+
+    def clean(self):
+        super().clean()
+
+        entity = getattr(self, self.entity_type)
+        preferred_aliases = entity.aliases.filter(
+            Q(preferred=True) & Q(id=entity.id)
+        )
+
+        if len(preferred_aliases) >= 1 and self.preferred:
```

Comment:
Will `preferred_aliases` ever be more than one? (Does this method fire before or after a new alias is created?)

Reply:
Depending on the answer to this question, you might be able to simplify the next couple of lines.